LabProject/src/main/java/agh/ics/oop/AbstractWorldMap.java

```diff
@@ -0,0 +1,46 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class AbstractWorldMap implements IWorldMap{
+    protected List<Animal> animalsOnField= new ArrayList<>();
+    protected List<Grass> grassOnField= new ArrayList<>();
+
+    @Override
+    public boolean isOccupied(Vector2d position) {return (objectAt(position)!=null);}
+
+    @Override
+    public Object objectAt(Vector2d position) {
+        for (Animal currAnimal : animalsOnField){
+            if(currAnimal.getPosition().equals(position)){
+                return currAnimal;
+            }
+        }
+        for (Grass currGrass : grassOnField){
```

Comment:
w konsekwencji powyższego - to powinno być w klasie `GrassField`, a tu możemy zostawić tylko część związaną zwierzętami